posthog/models/event/events_query.py

```diff
@@ -50,24 +50,40 @@ def run_events_query(
     offset = 0 if query.offset is None else query.offset
 
     # columns & group_by
-    select_input_raw = ["*"] if len(query.select) == 0 else query.select
-    select_input: List[str] = []
+    select_input = ["*"] if len(query.select) == 0 else query.select
     person_indices: List[int] = []
-    for index, col in enumerate(select_input_raw):
+
+    select: List[ast.Alias] = []
+    group_by: List[ast.Expr] = []
+    aggregations: List[ast.Expr] = []
+
+    for index, col in enumerate(select_input):
         # Selecting a "*" expands the list of columns, resulting in a table that's not what we asked for.
         # Instead, ask for a tuple with all the columns we want. Later transform this back into a dict.
         if col == "*":
-            select_input.append(f"tuple({', '.join(SELECT_STAR_FROM_EVENTS_FIELDS)})")
-        elif col.split("--")[0].strip() == "person":
+            select_expr = f"tuple({', '.join(SELECT_STAR_FROM_EVENTS_FIELDS)})"
+        elif col.split("--")[0].split(" as ")[0].strip() == "person":
```

Comment:
This `as` is the main change in this file. You can now alias `person as "whatever"` if you want to